File: src/leetcode/2026/01/Solution521.cpp
#include <string>
#include <iostream>
using namespace std;

class Solution521 {
public:
    int findLUSlength(string a, string b) {
        if (a == b) {
            return -1;
        } else {
            return max(a.size(), b.size());
        }
    }
};

int main() {
    Solution521 solution;

    string a1 = "aba";
    string b1 = "cdc";
    cout << solution.findLUSlength(a1, b1) << endl; // Expected output: 3

    string a2 = "aaa";
    string b2 = "aaa";
    cout << solution.findLUSlength(a2, b2) << endl; // Expected output: -1

    return 0;
}

File: src/leetcode/2026/01/Solution521.java
public class Solution521 {
    public int findLUSlength(String a, String b) {
        if (a.equals(b)) {
            return -1;
        }
        return Math.max(a.length(), b.length());
    }

    public static void main(String[] args) {
        Solution521 solution = new Solution521();

        String a1 = "aba";
        String b1 = "cdc";
        System.out.println("The length of the longest uncommon subsequence between '" + a1 + "' and '" + b1 + "' is: " +
                solution.findLUSlength(a1, b1));

        String a2 = "aaa";
        String b2 = "aaa";
        System.out.println("The length of the longest uncommon subsequence between '" + a2 + "' and '" + b2 + "' is: " +
                solution.findLUSlength(a2, b2));
    }
}
